### ISSUE TITLE
Crash at start, on a new install

### ISSUE THREAD

**Issue by Zalunda** [xbapps/xbvr#557]
It's the first time that I use XBVR (on Windows). 
I had nothing installed. 
When I started it the first time, I saw an icon in the taskbar, that disappeared when I hovered over it.
I tried a few more times and I always get the same result. I also tried to delete "%appdata%\xbvr" with the same results.
I'm not even able to get to the folder configuration part.

When I start it with "xbvr.exe >xbvrconsole.log 2>xbvrerror.log" on Windows, I get those logs:

"xbvrconsole.log":

time="2021-08-12T21:58:04-04:00" level=info msg="Migration did run successfully"
time="2021-08-12T21:58:04-04:00" level=info msg="Starting router "
time="2021-08-12T21:58:04-04:00" level=info msg="Added realm: default"

"xbvrerror.log":

panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xc0000005 code=0x0 addr=0x80 pc=0x8fc21d]

goroutine 417 [running]:
github.com/gammazero/nexus/v3/client.(*Client).Connected(...)
	/go/pkg/mod/github.com/gammazero/nexus/v3@v3.0.3/client/client.go:126
github.com/gammazero/nexus/v3/client.(*Client).Publish(0x0, 0x256fb64, 0xb, 0x0, 0x0, 0x0, 0x0, 0xc000b95230, 0x2769398, 0x0)
	/go/pkg/mod/github.com/gammazero/nexus/v3@v3.0.3/client/client.go:297 +0x2d
github.com/xbapps/xbvr/pkg/common.(*WampHook).Fire(0xc000b9c670, 0xc0013ba620, 0x4, 0xc00136c088)
	/go/appsrc/pkg/common/logging.go:36 +0x2c8
github.com/sirupsen/logrus.LevelHooks.Fire(0xc0000ad380, 0x4, 0xc0013ba620, 0xc0015fe880, 0xc001813cf0)
	/go/pkg/mod/github.com/sirupsen/logrus@v1.7.0/hooks.go:28 +0x98
github.com/sirupsen/logrus.(*Entry).fireHooks(0xc0013ba620)
	/go/pkg/mod/github.com/sirupsen/logrus@v1.7.0/entry.go:262 +0x8a
github.com/sirupsen/logrus.Entry.log(0x32006c0, 0xc000b951d0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/go/pkg/mod/github.com/sirupsen/logrus@v1.7.0/entry.go:237 +0x117
github.com/sirupsen/logrus.(*Entry).Log(0xc0013ba4d0, 0x4, 0xc001813e88, 0x1, 0x1)
	/go/pkg/mod/github.com/sirupsen/logrus@v1.7.0/entry.go:283 +0xf4
github.com/sirupsen/logrus.(*Entry).Logln(0xc0013ba4d0, 0xc000000004, 0xc000b95170, 0x3, 0x3)
	/go/pkg/mod/github.com/sirupsen/logrus@v1.7.0/entry.go:374 +0xda
github.com/sirupsen/logrus.(*Entry).Infoln(...)
	/go/pkg/mod/github.com/sirupsen/logrus@v1.7.0/entry.go:387
github.com/sirupsen/logrus.(*Entry).Println(...)
	/go/pkg/mod/github.com/sirupsen/logrus@v1.7.0/entry.go:391
github.com/sirupsen/logrus.(*Logger).Println(0x32006c0, 0xc000b95170, 0x3, 0x3)
	/go/pkg/mod/github.com/sirupsen/logrus@v1.7.0/logger.go:311 +0x71
github.com/gammazero/nexus/v3/router.(*router).Close.func1()
	/go/pkg/mod/github.com/gammazero/nexus/v3@v3.0.3/router/router.go:293 +0xbd
github.com/gammazero/nexus/v3/router.(*router).run(0xc000395ae0)
	/go/pkg/mod/github.com/gammazero/nexus/v3@v3.0.3/router/router.go:374 +0x50
created by github.com/gammazero/nexus/v3/router.NewRouter
	/go/pkg/mod/github.com/gammazero/nexus/v3@v3.0.3/router/router.go:93 +0x278

I also tried to use it in docker and it's not working either (web site is not available), but I'm less sure since I'm not even able to open a "cli" on the container to see what is going on, and I don't really know Docker that well.

**Comment by theRealKLH** [comment]
I just tried latest on Win10 and it worked fine (for me). you sure you are running the "Windows" version?

for docker... there is no "cli",  to get logs, the command is  "docker logs (name of container)".

**Comment by Zalunda** [comment]
I did more tests.

Tested on my laptop: it worked.
So there is something on my main computer that doesn't "like" XBVR.

I also tested old releases, going back to the first available one, and all the versions crash at the same place.

From what I can gather in the stack trace, it crashes when trying to create the WAMP router. 

	wampRouter, err := router.NewRouter(routerConfig, &log)

Somehow, it gets to "github.com/gammazero/nexus/v3/client.(*Client).Connected(...)" with "c = null" or "c.ctx = null" (or something).

![image](https://user-images.githubusercontent.com/8295521/129463577-3ff2729e-8632-4e38-a9bc-5bb7c711a999.png)

That been said, I don't mind anymore since I was able to make it work in Docker. I had to change "--net=host" for "-p 9998:9998 -p 9999:9999" to make it works on my computer.

**Comment by shushmo** [comment]
Same issue, crashing on start. 

Windows 10 Enterprise Version 20H2 OS Build 19042.1288.
Issue behavior:
Extracted the following version to drive where previous version exists: https://github.com/xbapps/xbvr/releases/download/0.3.17/xbvr_0.3.17_Windows_amd64.zip
Launched xbvr.exe from the new package. 
App launches, then quickly disappears from the task bar. Cannot open web UI.
Reverted back to old version.
Launched xbvr.exe
Continued without issue on version 0.3.16.

**Comment by theRealKLH** [comment]
> Same issue, crashing on start.
> 
> Windows 10 Enterprise Version 20H2 OS Build 19042.1288. Issue behavior: Extracted the following version to drive where previous version exists: https://github.com/xbapps/xbvr/releases/download/0.3.17/xbvr_0.3.17_Windows_amd64.zip Launched xbvr.exe from the new package. App launches, then quickly disappears from the task bar. Cannot open web UI. Reverted back to old version. Launched xbvr.exe Continued without issue on version 0.3.16.


Not the same issue. Your problem is due to VRBangers scraper issue.

**Comment by venturevoom** [comment]
just for the record - Win 10 Pro 64bit Build 19041 here, v 3.17 is crashing on start, 3.16 runs flawlessly. Would be awesome if this could be fixed. :D

**Comment by theRealKLH** [comment]
@Zalunda cleaning "dead" issues

**Comment by Zalunda** [comment]
Well, let's see if I comment on a closed issue (it might help someone else in the future). 

Recently, I tried again to use XBVR on my computer, instead of Docker, and I found a problem, that could have been the source all along. 

When I tried again, I didn't get an error message in the logs like before, I only got:

time="2023-03-27T12:06:09-04:00" level=info msg="Migration did run successfully"
time="2023-03-27T12:06:09-04:00" level=info msg="Starting router "
time="2023-03-27T12:06:09-04:00" level=info msg="Added realm: default"
And then XBVR would close.

If I ran docker at the same time, using the same ports, I got:

time="2023-03-27T12:07:59-04:00" level=info msg="Migration did run successfully"
time="2023-03-27T12:07:59-04:00" level=info msg="Starting router "
time="2023-03-27T12:07:59-04:00" level=info msg="Added realm: default"
time="2023-03-27T12:07:59-04:00" level=fatal msg="listen tcp 0.0.0.0:9998: bind: Only one usage of each socket address (protocol/network address/port) is normally permitted."
And then XBVR would close.

Which is normal. 

But I found another process on my machine was using port 9998 on my machine at the same time (using 'netstat -ab' as an administrator):

![image](https://user-images.githubusercontent.com/8295521/228098605-cf2dbe2e-0a9c-40fe-9d6c-ed8f2411eb00.png)

It was starting a node.js server on port 9998 also but, I think, on http instead of tcp. 

Long story, short. When I uninstalled it, I was able to use XBVR without any issue. 
I'm not sure why it was causing problems for XBVR, without triggering the "Only one usage of each socket address" or something like that. Maybe since it was not the same protocol, it could be reserved at the same time but didn't works.
